daemon/algod/api/server/v2/dryrun.go

```diff
@@ -182,23 +182,23 @@ func (ddr *dryrunDebugReceiver) stateToState(state *logic.DebugState) generated.
 }
 
 // Register is fired on program creation (DebuggerHook interface)
-func (ddr *dryrunDebugReceiver) Register(state *logic.DebugState) error {
+func (ddr *dryrunDebugReceiver) BeforeAppEval(state *logic.DebugState) error {
 	ddr.disassembly = state.Disassembly
 	ddr.lines = strings.Split(state.Disassembly, "\n")
 	return nil
 }
 
 // Update is fired on every step (DebuggerHook interface)
```

Comment:
nit: function names in comments are outdated.

Also, the behavior of this `dryrunDebugReceiver` will now change, since `BeforeTealOp` will be invoked for inner txn ops as well? And previously it was not?

Reply:
It previously was as well. For the most part, `BeforeTealOp` is a renaming of `Update`. See https://github.com/jdtzmn/go-algorand/pull/4/files#diff-7efbb393ce12c86b88888ff627dd385ba50ec9937b26082263ef580068080d27L722-R722.

Reply:
Previously `NewInnerEvalParams` did not propagate the `Debugger` property from the parent `EvalParams`, and in this PR it does. So I think the current dryrun would see a different behavior here.

Reply:
That's a great point. I hadn't thought of that. I'm assuming this new behavior is something we don't want. A simple solution would be to have a flag which disables the teal op logic when within an inner transaction. Do you think that would be sufficient?

Reply:
Yep, that sounds like a good plan